otherlibs/systhreads/st_posix.h

```diff
@@ -356,19 +356,18 @@ static void st_decode_sigset(value vset, sigset_t * set)
 
 static value st_encode_sigset(sigset_t * set)
 {
-  value res = Val_int(0);
+  CAMLparam0();
+  CAMLlocal1(res);
   int i;
 
-  Begin_root(res)
-    for (i = 1; i < NSIG; i++)
-      if (sigismember(set, i) > 0) {
-        value newcons = caml_alloc_small(2, 0);
-        Field(newcons, 0) = Val_int(caml_rev_convert_signal_number(i));
-        Field(newcons, 1) = res;
-        res = newcons;
-      }
-  End_roots();
-  return res;
+  for (i = 1; i < NSIG; i++)
```

Comment:
I would prefer an explicit initialization to the empty list before the loop: `res = Val_int(0);` .  `CAMLlocal1` does it already, but it's not obvious.

Reply:
Fixed, thanks.

Reply:
Old school's cool but could `Val_emptylist` be used instead?

Reply:
Wish granted.